code/modules/surgery/organs/wings.dm

```diff
@@ -145,9 +145,61 @@
 	name = "pair of bee wings"
 	desc = "A pair of bee wings. They seem tiny and undergrown"
 	icon_state = "beewings"
-	flight_level = WINGS_FLIGHTLESS
+	flight_level = WINGS_COSMETIC
+	actions_types = list(/datum/action/item_action/organ_action/use/bee_dash)
 	wing_type = "Bee"
 
+/datum/action/item_action/organ_action/use/bee_dash
+	var/jumpdistance = 3
+	var/jumpspeed = 1
+	var/recharging_rate = 100
+	var/recharging_time = 0
+
+/datum/action/item_action/organ_action/use/bee_dash/Trigger()
+	var/mob/living/carbon/L = owner
+
+	if(L.stat != CONSCIOUS || L.buckling) // Has to be concious and unbuckled
+		return
+	if(recharging_time > world.time)
+		to_chat(L, "<span class='warning'>The wings aren't ready to dash yet!</span>")
+		return
+	var/datum/gas_mixture/environment = L.loc.return_air()
+	if(environment && !(environment.return_pressure() > 30))
+		to_chat(L, "<span class='warning'>The atmosphere is too thin for you to dash!</span>")
+		return
+
+	var/turf/target = get_edge_target_turf(L, L.dir) //represents the user's direction
+	var/hoppingtable = FALSE // Triggers the trip
+	var/jumpdistancemoved = jumpdistance // temp jumpdistance
+	var/turf/checkjump = get_turf(L)
+
+	check_loop:
+		for(var/i in 1 to jumpdistance) //This is how hiero club find the tiles in front of it, tell me/fix it if there's a better way
+			var/turf/T = get_step(checkjump, L.dir)
+			if(locate(/obj/structure/table) in T) // If there's a table, trip
+				hoppingtable = TRUE
+				jumpdistancemoved = i
+				break
+			if(T.density) // If turf has density (wall), break
+				break
+			for(var/atom/movable/A in T) // breaks if there's anything solid in the way, no tripping on tables through walls
+				if(A.density == TRUE)
+					break check_loop
+			checkjump = get_step(checkjump, L.dir)
```

Comment:
this is kinda shitcodey, use a normal loop without the goto tag things

Reply:
```suggestion
	for(var/i in 1 to jumpdistance) //This is how hiero club find the tiles in front of it, tell me/fix it if there's a better way
		var/turf/T = get_step(checkjump, L.dir)
		if(locate(/obj/structure/table) in T) // If there's a table, trip
			hoppingtable = TRUE
			umpdistancemoved = i
			break
		if(!checkjump.Adjacent(T)) // Check for things other than tables that would block flight
			break
```

Ideally, we should add check for tables at the throwing code, but we all know that it will be a mess without much architectural changes. This suggestion is some sort of a compromise that'll work okay-ish.

Reply:
goto tags, wew

Reply:
after testing, this does not work

https://user-images.githubusercontent.com/63219222/105901378-497fd600-5feb-11eb-86d4-ddd5a63ff8b0.mp4